pallas-txbuilder/src/transaction/model.rs

```diff
@@ -340,8 +340,41 @@ impl StagingTransaction {
+        let plutus_v3_costmodel: Vec<u8> = vec![161, 1, 152, 251, 26, 0, 1, 137, 180, 25, 1, 164, 1, 1, 25, 3, 232, 24, 173, 0, 1, 25, 3, 232, 25, 234, 53, 4, 1, 25, 43, 175, 24, 32, 26, 0, 3, 18, 89, 25, 32, 164, 4, 25, 62, 128, 24, 100, 25, 62, 128, 24, 100, 25, 62, 128, 24, 100, 25, 62, 128, 24, 100, 25, 62, 128, 24, 100, 25, 62, 128, 24, 100, 24, 100, 24, 100, 25, 62, 128, 24, 100, 26, 0, 1, 112, 167, 24, 32, 26, 0, 2, 7, 130, 24, 32, 25, 240, 22, 4, 26, 0, 1, 25, 74, 24, 178, 0, 1, 25, 86, 135, 24, 32, 26, 0, 1, 100, 53, 25, 3, 1, 4, 2, 26, 0, 1, 79, 88, 26, 0, 1, 225, 67, 25, 28, 137, 57, 3, 131, 25, 6, 180, 25, 2, 37, 24, 57, 26, 0, 1, 79, 88, 0, 1, 1, 25, 3, 232, 25, 167, 169, 4, 2, 25, 95, 228, 25, 115, 58, 24, 38, 1, 26, 0, 13, 180, 100, 25, 106, 143, 1, 25, 202, 63, 25, 2, 46, 1, 25, 153, 16, 25, 3, 232, 25, 236, 178, 1, 26, 0, 2, 42, 71, 24, 32, 26, 0, 1, 68, 206, 24, 32, 25, 59, 195, 24, 32, 26, 0, 1, 41, 17, 1, 25, 51, 113, 4, 25, 86, 84, 10, 25, 113, 71, 24, 74, 1, 25, 113, 71, 24, 74, 1, 25, 169, 21, 25, 2, 40, 1, 25, 174, 205, 25, 2, 29, 1, 25, 132, 60, 24, 32, 26, 0, 1, 10, 150, 24, 32, 26, 0, 1, 26, 170, 24, 32, 25, 28, 75, 24, 32, 25, 28, 223, 24, 32, 25, 45, 26, 24, 32, 26, 0, 1, 79, 88, 26, 0, 1, 225, 67, 25, 28, 137, 57, 3, 131, 25, 6, 180, 25, 2, 37, 24, 57, 26, 0, 1, 79, 88, 0, 1, 26, 0, 1, 97, 66, 25, 2, 7, 0, 1, 26, 0, 1, 34, 193, 24, 32, 26, 0, 1, 79, 88, 26, 0, 1, 225, 67, 25, 28, 137, 57, 3, 131, 25, 6, 180, 25, 2, 37, 24, 57, 26, 0, 1, 79, 88, 0, 1, 1, 26, 0, 1, 79, 88, 26, 0, 1, 225, 67, 25, 28, 137, 57, 3, 131, 25, 6, 180, 25, 2, 37, 24, 57, 26, 0, 1, 79, 88, 0, 1, 26, 0, 14, 148, 114, 26, 0, 3, 65, 64, 0, 2, 26, 0, 4, 33, 60, 25, 88, 60, 4, 26, 0, 22, 60, 173, 25, 252, 54, 4, 25, 79, 243, 1, 4, 0, 26, 0, 2, 42, 168, 24, 32, 26, 0, 1, 137, 180, 25, 1, 164, 1, 1, 26, 0, 1, 62, 255, 24, 32, 25, 232, 106, 24, 32, 25, 78, 174, 24, 32, 25, 96, 12, 24, 32, 25, 81, 8, 24, 32, 25, 101, 77, 24, 32, 25, 96, 47, 24, 32, 26, 2, 144, 241, 231, 10, 26, 3, 46, 147, 175, 25, 55, 253, 10, 26, 2, 152, 228, 11, 25, 102, 196, 10, 25, 62, 128, 24, 100, 25, 62, 128, 24, 100, 26, 0, 14, 175, 31, 18, 26, 0, 42, 110, 6, 6, 26, 0, 6, 190, 152, 1, 26, 3, 33, 170, 199, 25, 14, 172, 18, 26, 0, 4, 22, 153, 18, 26, 4, 142, 70, 110, 25, 34, 164, 18, 26, 3, 39, 236, 154, 18, 26, 0, 30, 116, 60, 24, 36, 26, 0, 49, 65, 15, 12, 26, 0, 13, 191, 158, 1, 26, 9, 242, 246, 211, 25, 16, 211, 24, 36, 26, 0, 4, 87, 130, 24, 36, 26, 9, 110, 68, 2, 25, 103, 181, 24, 36, 26, 4, 115, 206, 232, 24, 36, 26, 19, 230, 36, 114, 1, 26, 15, 35, 212, 1, 24, 72, 26, 0, 33, 44, 86, 24, 72, 26, 0, 34, 129, 70, 25, 252, 59, 4, 26, 0, 3, 43, 0, 25, 32, 118, 4, 26, 0, 19, 190, 4, 25, 112, 44, 24, 63, 0, 1, 26, 0, 15, 89, 217, 25, 170, 103, 24, 251, 0, 1];
+
+        let mut buf = Vec::new();
+        if redeemers.len() == 0 && datums.is_some() {
+            buf.push(0xA0);
```

Comment:
This is the value that changed from `0x80` -> `0xA0` for Conway Era, correct?

Reply:
yes, and is just a side effect of encoding an empty map of redeemers.

To correctly handle this, the different era implementations of `redeemers` should have era-specific serialization code; because Chang+1 will mean the redeemers now must be encoded as a map.

Reply:
instead of handling cbor bytes directly, it would be easier / safer to have a struct `ScriptDataHash` that implements cbor encoding and contains the underlying data parts (empty maps and such nuances would be handled by the data type).

This struct could live inside primitives crate and get versioned by era (`babbage:ScriptDataHash`, `conway:ScriptDataHash`, etc).